internal/infrastructure/kubernetes/metrics.go

```diff
@@ -0,0 +1,21 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package kubernetes
+
+import "github.com/envoyproxy/gateway/internal/metrics"
+
+var (
+	infraCreateOrUpdateFailed          = metrics.NewCounter("infra_create_or_update_failed", "Number of created or updated infrastructures that failed.")
```

Comment:
* can we split up `create` from `update` metrics ?
* can we use another prefix like `resource` instead of infra

Reply:
seems a little tricky to split `create` from `update`, cause they are using the same code to manage resources, 

https://github.com/envoyproxy/gateway/blob/0177dac7a31065eaeb673aa04c751f0c7ce651b7/internal/infrastructure/kubernetes/infra_client.go#L27

may directly using `apply` sounds better than `createOrUpdate` ?

Reply:
change name from `createOrUpdate` to `apply`, and change the prefix of metrics to `resource`